wallets/react-wallet-v2/src/lib/TezosLib.ts

```diff
@@ -0,0 +1,109 @@
+import { TezosToolkit } from '@taquito/taquito'
+import { InMemorySigner } from '@taquito/signer'
+import { localForger } from '@taquito/local-forging'
+import Keyring from 'mnemonic-keyring'
+
+/**
+ * Constants
+ */
+const DEFAULT_PATH = "m/44'/1729'/0'/0'"
+const DEFAULT_CURVE = 'ed25519'
+
+/**
+ * Types
+ */
+interface IInitArguments {
+  mnemonic?: string
+  path?: string
+  curve?: 'ed25519' | 'secp256k1'
+}
+
+/**
+ * Library
+ */
+export default class TezosLib {
+  tezos: TezosToolkit
+  signer: InMemorySigner
+  mnemonic: string
+  secretKey: string
+  publicKey: string
+  address: string
+  curve: 'ed25519' | 'secp256k1'
+
+  constructor(
+    tezos: TezosToolkit,
+    mnemonic: string,
+    signer: InMemorySigner,
+    secretKey: string,
+    publicKey: string,
+    address: string,
+    curve: 'ed25519' | 'secp256k1'
+  ) {
+    this.tezos = tezos
+    this.mnemonic = mnemonic
+    this.signer = signer
+    this.secretKey = secretKey
+    this.publicKey = publicKey
+    this.address = address
+    this.curve = curve
+  }
+
+  static async init({ mnemonic, path, curve }: IInitArguments) {
+    const params = {
+      mnemonic: mnemonic ?? Keyring.generateMnemonic(),
+      derivationPath: path ?? DEFAULT_PATH,
+      curve: curve ?? DEFAULT_CURVE
+    }
+
+    const Tezos = new TezosToolkit('https://mainnet.api.tez.ie')
```

Comment:
So there is no way to `init` to a testnet? Is this intentional?

Reply:
+1, additionally I seem to get the following `500` when I try to perform `tezos_send`. Could of course just be because I don't have funds in the wallet Tezos mainnet but 500 seems like an odd return code for that:
<img width="589" alt="Screenshot 2023-03-27 at 20 47 55" src="https://user-images.githubusercontent.com/8663099/228038075-958ffe1d-2975-4765-90bc-a49d1a3ad858.png">

Reply:
I quickly checked, I did not find how I can easily use the rpc for the correct network that is defined in `TezosData.ts`. It seems that other chains don't do this either.